src/Datasource/Paginator.php

```diff
@@ -220,7 +220,10 @@ public function paginate($object, array $params = [], array $settings = [])
         $this->_pagingParams = [$alias => $paging];
 
         if ($requestedPage > $page) {
-            throw new PageOutOfBoundsException([$requestedPage]);
+            throw new PageOutOfBoundsException([
+                'requestedPage' => $requestedPage,
+                'pagingParams' => $this->_pagingParams
```

Comment:
Yes, but could there be multiple keys in here? How would you then know to extract the correct alias etc?